crates/environ/src/compilation.rs

```diff
@@ -363,8 +363,8 @@ pub trait Compiler: Send + Sync {
         funcs: &PrimaryMap<DefinedFuncIndex, (SymbolId, &(dyn Any + Send))>,
     ) -> Result<()>;
 
-    /// The function alignment required by this ISA.
-    fn function_alignment(&self) -> u32;
+    /// The minimum function alignment required by this ISA.
+    fn min_function_alignment(&self) -> u32;
```

Comment:
What would you say to removing this method from the `compiler` trait entirely? If it's only used for `append_func`, which already has an `alignment` parameter, maybe we should just pass the desired alignment at `append_func`'s callers.